tests/org.jboss.tools.docker.ui.bot.test/src/org/jboss/tools/docker/ui/bot/test/container/VolumeMountTest.java

```diff
@@ -11,73 +11,73 @@
-	private String containerName = "test_run_wildfly_volumes";
-	private String deploymentPath = "resources/wildfly-deployments";
-	private String containerPath = "/opt/jboss/wildfly/standalone/deployments/";
-	private String quickstartURL = "wildfly-helloworld";
-
+	private static final String IMAGE_NAME = "fnichol/uhttpd";
+	private static final String IMAGE_TAG = "latest";
+	private static final String CONTAINER_NAME = "test_run_volumes";
+	private static final String VOLUME_PATH = "resources/test-volumes";
+	private static final String CONTAINER_PATH = "/www";
+	private static final String INDEX_PAGE = "index.html";
+	private static final String INDEX_PAGE_PATH = VOLUME_PATH + "/" + INDEX_PAGE;
+	private static final String HOST_PORT = "80";
+	
+	@Before
+	public void before() {
+		deleteContainerIfExists(CONTAINER_NAME);
+		pullImage(IMAGE_NAME, IMAGE_TAG);
+	}
+	
 	@Test
 	public void testVolumeMount() throws IOException {
-		pullImage(imageName, imageTag);
-		DockerImagesTab imageTab = new DockerImagesTab();
-		imageTab.activate();
-		imageTab.refresh();
-		new WaitWhile(new JobIsRunning());
-		imageTab.runImage(imageName + ":" + imageTag);
+		DockerImagesTab imagesTab = openDockerImagesTab();
+		imagesTab.runImage(IMAGE_NAME + ":" + IMAGE_TAG);
+
 		ImageRunSelectionPage firstPage = new ImageRunSelectionPage();
-		firstPage.setName(this.containerName);
-		firstPage.setPublishAllExposedPorts(false);
+		firstPage.setName(CONTAINER_NAME);
+		firstPage.setPublishAllExposedPorts(true);
 		firstPage.next();
+		
 		ImageRunResourceVolumesVariablesPage secondPage = new ImageRunResourceVolumesVariablesPage();
-		String deploymentPath = "";
-		try {
-			deploymentPath = (new File(this.deploymentPath)).getCanonicalPath();
-		} catch (IOException ex) {
-			fail("Resource file not found!");
-		}
-		secondPage.addDataVolumeToHost(containerPath, deploymentPath);
+		String volumePath = (new File(VOLUME_PATH)).getCanonicalPath();
+		secondPage.addDataVolumeToHost(CONTAINER_PATH, volumePath);
 		secondPage.finish();
 		new WaitWhile(new JobIsRunning());
 		new WaitWhile(new ConsoleHasNoChange());
-		String url = createURL(":8080" + "/" + quickstartURL);
-		BrowserView browserView = new BrowserView();
-		browserView.open();
-		browserView.openPageURL(url);
-		DeployOnServer.checkBrowserForErrorPage(browserView, url);
 
+		String indexPage = getIndexPageContent();
+		String indexPageResource = getResourceAsString(INDEX_PAGE_PATH);
+		assertEquals(INDEX_PAGE_PATH + " wasnt mounted/displayed properly.", indexPage, indexPageResource);
 	}
 
-	@After
-	public void after() {
-		deleteContainer(this.containerName);
-		deleteImage(imageName, imageTag);
+	private String getIndexPageContent() {
+		String containerIP = getContainerIP(CONTAINER_NAME);
+		String url = "http://" + containerIP + ":" + HOST_PORT + "/" + INDEX_PAGE;
+		BrowserView browserView = new BrowserView();
+		browserView.open();
+		browserView.openPageURL(url);
+		return browserView.getText();
 	}
-
+	
```

Comment:
I agree on removing the container but I dont think that we should remove images, unless we modified them. It turns out that this spares a lot of time (they dont have to get re-pulled).